app/scripts/metamask-controller.js

```diff
@@ -2183,6 +2187,15 @@ export default class MetamaskController extends EventEmitter {
         },
         this.controllerMessenger,
       ),
+      processRequestExecutionPermissions:
```

Comment:
after this feature is enabled and shipped, I'd love to rename this hook to plain old `handleSnapRequest` or something like that

Reply:
Great idea! I have updated this to `forwardRequestToSnap` which I think is pretty clear about what it's doing. Presently it forwards the request to the specified snap verbatim, but in the future we could support a `mapRequest` function that could accept the incoming request and return the snap request, so that the publicly exposed method doesn't necessarily need to match exactly the snap method.

https://github.com/MetaMask/metamask-extension/pull/35193/files#diff-6fbff2cfe97ac01b77296ef2122c7e0a5b3ff6a84b584b4d1a87482f35eea3d6R2192

lmkwyt